aicsimageio/readers/reader.py

```diff
@@ -680,6 +680,37 @@ def channel_names(self) -> Optional[List[str]]:
 
         return None
 
+    @staticmethod
+    def _generate_coord_array(
+        start: Union[int, float], stop: Union[int, float], step_size: Union[int, float]
+    ) -> np.ndarray:
+        """
+        Generate an np.ndarray for coordinate values.
+
+        Parameters
+        ----------
+        start: Union[int, float]
+            The start value.
+        stop: Union[int, float]
+            The stop value.
+        step_size: Union[int, float]
+            How large each step should be.
+
+        Returns
+        -------
+        coords: np.ndarray
+            The coordinate array.
+
+        Notes
+        -----
+        In general, we have learned that floating point math is hard....
+        This block of code used to use `np.arange` with floats as parameters and
+        it was causing errors. To solve, we generate the range with ints and then
+        multiply by a float across the entire range to get the proper coords.
+        See: https://github.com/AllenCellModeling/aicsimageio/issues/249
+        """
+        return np.arange(start, stop) * step_size
```

Comment:
~30 lines of code for 1 line of actual logic 😂 Amazing, really well done and thorough!!